src/modules/cpu/host_fused_functions.hpp

```diff
@@ -4,12 +4,593 @@
+        //         //                 | [6],  [7],  [8], [X] |  (R/G/B/X values for pixel n + 2)                                | [2], [6], [10], [14] | (stored in pTransposedCtmRow2)
+        //         //                 | [9], [10], [11], [X] |  (R/G/B/X values for pixel n + 3)                                | [3], [7], [11], [15] | (stored in pTransposedCtmRow3)
+        //         //                  --                  --                                                                    --                  --
+        //         // This matrix multiplication gives results for 4 pixels in one shot - each row of the result matrix containing one set of R/G/B/X values.
+
+        //         // Load
+
+        //         px0 =  _mm_loadu_si128((__m128i *)srcPtrTemp);
+
+        //         // Unpack u8 to f32
+
+        //         px1 = _mm_unpackhi_epi8(px0, zero);    // pixels 8-15
+        //         px0 = _mm_unpacklo_epi8(px0, zero);    // pixels 0-7
+        //         p0 = _mm_cvtepi32_ps(_mm_unpacklo_epi16(px0, zero));    // pixels 0-3
+        //         p1 = _mm_cvtepi32_ps(_mm_unpackhi_epi16(px0, zero));    // pixels 4-7
+        //         p2 = _mm_cvtepi32_ps(_mm_unpacklo_epi16(px1, zero));    // pixels 8-11
+        //         p3 = _mm_cvtepi32_ps(_mm_unpackhi_epi16(px1, zero));    // pixels 12-15
+
+        //         // Shuffle to broadcast
+
+        //         pSrcF32[0] = _mm_shuffle_ps(p0, p0, _MM_SHUFFLE(0,0,0,0)); // pix0R, pix0R, pix0R, pix0R
+        //         pSrcF32[1] = _mm_shuffle_ps(p0, p0, _MM_SHUFFLE(1,1,1,1)); // pix0G, pix0G, pix0G, pix0G
+        //         pSrcF32[2] = _mm_shuffle_ps(p0, p0, _MM_SHUFFLE(2,2,2,2)); // pix0B, pix0B, pix0B, pix0B
+        //         pSrcF32[3] = _mm_shuffle_ps(p0, p0, _MM_SHUFFLE(3,3,3,3)); // pix1R, pix1R, pix1R, pix1R
+
+        //         pSrcF32[4] = _mm_shuffle_ps(p1, p1, _MM_SHUFFLE(0,0,0,0)); // pix1G, pix1G, pix1G, pix1G
+        //         pSrcF32[5] = _mm_shuffle_ps(p1, p1, _MM_SHUFFLE(1,1,1,1)); // pix1B, pix1B, pix1B, pix1B
+        //         pSrcF32[6] = _mm_shuffle_ps(p1, p1, _MM_SHUFFLE(2,2,2,2)); // pix2R, pix2R, pix2R, pix2R
+        //         pSrcF32[7] = _mm_shuffle_ps(p1, p1, _MM_SHUFFLE(3,3,3,3)); // pix2G, pix2G, pix2G, pix2G
+
+        //         pSrcF32[8] = _mm_shuffle_ps(p2, p2, _MM_SHUFFLE(0,0,0,0)); // pix2B, pix2B, pix2B, pix2B
+        //         pSrcF32[9] = _mm_shuffle_ps(p2, p2, _MM_SHUFFLE(1,1,1,1)); // pix3R, pix3R, pix3R, pix3R
+        //         pSrcF32[10] = _mm_shuffle_ps(p2, p2, _MM_SHUFFLE(2,2,2,2)); // pix3G, pix3G, pix3G, pix3G
+        //         pSrcF32[11] = _mm_shuffle_ps(p2, p2, _MM_SHUFFLE(3,3,3,3)); // pix3B, pix3B, pix3B, pix3B
+
+        //         for(int i = 0, j = 0; i < 12; i+=3, j++)
+        //         {
+        //             pRowArr[j] = _mm_fmadd_ps(pSrcF32[i + 0],
+        //                                       pTransposedCtmRow0,
```

Comment:
You might want to try using mul_ps and add_ps instead

Reply:
I noticed that the latencies for mul_ps, add_ps, were 4 + 4 = 8, but fmadd_ps was only 4. But this was in the intel intrinsics guide please let me know if this varies.

Reply:
Ok. Let us revisit this after convergence